pkg/corpusaccess/corpusaccessor.py

```diff
@@ -0,0 +1,42 @@
+import yaml
+
+
+class CorpusAccessor:
+	# private construction for singleton
+	class __CorpusAccessor:
+		def __init__(self, ctx):
+			self.ctx = ctx
```

Comment:
Should we reference the URL source where we got this code from?

Reply:
sure I can add a link in a comment, I was following the first example here: https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Singleton.html